src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -251,7 +251,7 @@ public override BoundNode VisitReturnStatement(BoundReturnStatement node)
             {
                 var expression = node.ExpressionOpt;
                 var type = (expression is null) ?
-                    NoReturnExpression :
+                    TypeSymbolWithAnnotations.CreateUnannotated(NonNullTypesFalseContext.Instance, NoReturnExpression) :
```

Comment:
>NonNullTypesFalseContext.Instance [](start = 64, length = 33)

Feels like we should use the context from the containing method/symbol.

Reply:
`NoReturnExpression` is used as a placeholder only. And the `TypeSymbol` field is the only field used.

---
In reply to: [206364611](https://github.com/dotnet/roslyn/pull/28943#discussion_r206364611) [](ancestors = 206364611)

Reply:
Added `NonNullTypesUnusedContext`.

---
In reply to: [206540166](https://github.com/dotnet/roslyn/pull/28943#discussion_r206540166) [](ancestors = 206540166,206364611)